foundation/org.eclipse.persistence.core/src/org/eclipse/persistence/internal/helper/WriteLockManager.java

```diff
@@ -98,7 +102,9 @@ public Map acquireLocksForClone(Object objectForClone, ClassDescriptor descripto
                             toWaitOn.wait();// wait for lock on object to be released
                         }
                     } catch (InterruptedException ex) {
-                        // Ignore exception thread should continue.
+                        //https://jira.site1.hyperwallet.local/browse/HW-53073
+                        //Custom change to allow thread interruptions for bad threads stuck in org.eclipse.persistence.internal.helper.WriteLockManager.acquireLocksForClone pattern
```

Comment:
Can you add a timer and logger that indicates how long it was waiting before it was interrupted? The fact that it is being interrupted from somewhere other than your EclipseLinkLockedThreadHandler makes me worry that these interrupts will be way too frequent on PROD. Once we understand how old the wait is, we may decide to only throw the exception when wait is long or by somehow determining if EclipseLinkLockedThreadHandler is the interrupter.

Reply:
Also log 'tries'

Reply:
done.